projects/ngneat/helipopper/src/lib/intersection-observer.ts

```diff
@@ -0,0 +1,9 @@
+// Let's retrieve the native `IntersectionObserver` implementation hidden by
+// `__zone_symbol__IntersectionObserver`. This would be the unpatched version of
+// the observer present in zone.js environments.
+// Otherwise, if the user is using zoneless change detection (and zone.js is not included),
+// we fall back to the native implementation. Accessing the native implementation
+// allows us to remove `runOutsideAngular` calls and reduce indentation,
+// making the code a bit more readable.
+export const IntersectionObserver: typeof globalThis.IntersectionObserver =
```

Comment:
but what about applications that uses zone and need to run it outside the zone?

Reply:
That would get the first operand immediately (`globalThis.__zone_symbol__IntersectionObserver`).

Reply:
I meant that you removed the run outside zone call from the code

Reply:
Yeah it's not required. It's always running the native implementation (which is not intercepted by Angular).
- when zone.js is included it gets `__zone_symbol__IntersectionObserver`
- when zoneless is enabled it gets ` || .IntersectionObserver`

Reply:
But with zone js enabled [this](https://github.com/ngneat/helipopper/pull/159/files#diff-cdd940f8a1ed11596e83dea338ab6b8e0e20e3d39f7e0bfb10697d336a074cafL160) will run inside the zone

Reply:
It will. But the emission from the observable (`inView()`) would be outside of the zone. Native implementation is always running outside of the zone.

Reply:
> But the emission from the observable (inView()) would be outside of the zone

even if we use zone js?

Reply:
Yep:

![image](https://github.com/user-attachments/assets/4d6084be-a6e1-4736-aa87-77e04904ac39)

![image](https://github.com/user-attachments/assets/7a83de1b-8ece-40e9-8e5b-7722c9d84906)